### ISSUE TITLE
add warning state to text inputs

### ISSUE THREAD

**Issue by christianvogt** [patternfly/patternfly-react#4194]
**Describe the issue. What is the expected and unexpected behavior?**

PF text inputs can be marked invalid which then sets `aria-invalid="true"` and visually adds an error icon background and red underline.

We have a need to show a warning indicator. The reason for the warning is because the value cannot be detected as an error case but we have detected that the value may require some additional steps by the user.

Design mock up:
![image](https://user-images.githubusercontent.com/14068621/81190854-599be500-8f86-11ea-8fd9-1233d050c9c7.png)

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
enhancement

**What is your product and what release version are you targeting?**
Openshift Console 4.5+

**Comment by tlabaj** [comment]
cc @mcarrano

**Comment by mcarrano** [comment]
I'm OK to add this state. Do you see any issues @mceledonia ? Should probably start as a Core update.

**Comment by mceledonia** [comment]
Agreed, I think it can go to core. The warning icon looks a little big in the mock up above, but in implementation it should be correct as long as it's the same size we use for the error state.

**Comment by christianvogt** [comment]
bump

**Comment by mcarrano** [comment]
I am going to transfer this issue over to Core and get it on the roadmap. What timeframe are you looking for this is @christianvogt ?
